### PR TITLE
disable-foxx 3.4

### PR DESCRIPTION
**Pull request by ObiWahn**
Internal Ref:
- https://github.com/arangodb/planning/issues/3680

### CODE REVIEW

**Comment by jsteemann** [comment]
This PR is incomplete.
The variables `isFoxxStoreDisabled` and `isFoxxApiDisabled` are never read from.

**Comment by ObiWahn** [comment]
> This PR is incomplete.
> The variables `isFoxxStoreDisabled` and `isFoxxApiDisabled` are never read from.

Yes it is WIP.

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5481/

File: arangod/GeneralServer/ServerSecurityFeature.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      _enableFoxxStore(true) {
+  setOptional(false);
+  startsAfter("ServerPlatform");
+}
+
+void ServerSecurityFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
+  options->addSection("foxx", "Configure Foxx");
+  options->addOption("--foxx.api", "enables Foxx management REST APIs",
+                     new BooleanParameter(&_enableFoxxApi))
+                     .setIntroducedIn(30500);
```
Comment (jsteemann) [inline arangod/GeneralServer/ServerSecurityFeature.cpp]: this is not true anymore.
```suggestion
                     .setIntroducedIn(30408).setIntroducedIn(30500);
```

File: arangod/GeneralServer/ServerSecurityFeature.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  options->addOption("--foxx.store", "enables Foxx store in web interface",
+                     new BooleanParameter(&_enableFoxxStore))
+                     .setIntroducedIn(30500);
```
Comment (jsteemann) [inline arangod/GeneralServer/ServerSecurityFeature.cpp]: not true anymore
```suggestion
                     .setIntroducedIn(30408).setIntroducedIn(30500);
```

File: arangod/RestHandler/RestStatusHandler.cpp [hunk 3]
```diff
@@ -50,6 +51,17 @@ RestStatusHandler::RestStatusHandler(GeneralRequest* request, GeneralResponse* r
     : RestBaseHandler(request, response) {}
 
 RestStatus RestStatusHandler::execute() {
+  ServerSecurityFeature* security =
+      application_features::ApplicationServer::getFeature<ServerSecurityFeature>(
+          "ServerSecurity");
+  TRI_ASSERT(security != nullptr);
+
+  if (!security->canAccessHardenedApi()) {
```
Comment (jsteemann) [inline arangod/RestHandler/RestStatusHandler.cpp]: this is not necessary for the Foxx API toggles.

Comment (jsteemann) [reply arangod/RestHandler/RestStatusHandler.cpp]: Additionally, the method `canAccessHardenedApi` is not a member of the `ServerSecurityFeature` class above.

**Comment by jsteemann** [comment]
Additionally, this change needs CHANGELOG entries.

File: arangod/RestHandler/RestStatusHandler.cpp (hunk 3 above)
Comment (ObiWahn) [reply arangod/RestHandler/RestStatusHandler.cpp]: should not have made it to github

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/7130/

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5486/

File: CHANGELOG [hunk 4]
```diff
@@ -1,6 +1,12 @@
 v3.4.8 (XXXX-XX-XX)
 -------------------
 
+* Added options to make server more secure:
+  - `--foxx.api`: set to false disables Foxx management API
+  - `--foxx.store`: set to false disables Foxx UI
+
+* Fixed issue #9558: RTRIM not working as expected
```
Comment (jsteemann) [inline CHANGELOG]: The "RTRIM" entry is now a duplicate entry in the CHANGELOG

File: arangod/GeneralServer/ServerSecurityFeature.cpp [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      _enableFoxxStore(true) {
+  setOptional(false);
+  startsAfter("ServerPlatform");
+}
+
+void ServerSecurityFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
+  options->addSection("foxx", "Configure Foxx");
+  options->addOption("--foxx.api", "enables Foxx management REST APIs",
+                     new BooleanParameter(&_enableFoxxApi))
+                     .setIntroducedIn(30408);
```
Comment (jsteemann) [inline arangod/GeneralServer/ServerSecurityFeature.cpp]:
```suggestion
                     .setIntroducedIn(30408).setIntroducedIn(30500);
```

File: arangod/GeneralServer/ServerSecurityFeature.cpp [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  options->addOption("--foxx.store", "enables Foxx store in web interface",
+                     new BooleanParameter(&_enableFoxxStore))
+                     .setIntroducedIn(30408);
```
Comment (jsteemann) [inline arangod/GeneralServer/ServerSecurityFeature.cpp]:
```suggestion
                     .setIntroducedIn(30408).setIntroducedIn(30500);
```

**Comment by ObiWahn** [comment]
https://github.com/arangodb/docs/pull/126

**Pull request merged**